juicer/scikit_learn/etl_operation.py

```diff
@@ -15,15 +15,15 @@ class AddColumnsOperation(Operation):
     def __init__(self, parameters, named_inputs, named_outputs):
         Operation.__init__(self, parameters,  named_inputs, named_outputs)
 
-        self.has_code = len(named_inputs) == 2
+        self.has_code = len(named_outputs) == 1 and len(named_inputs) == 2
```

Comment:
Most outputs has a "many" multiplicity, i.e., it can be connected to multiple inputs. So, the test must be:
```
len(named_outputs) >= 1 and len(named_inputs) == 2
```

Reply:
Ok, I'll remember this :>